### DISCUSSION TITLE
Call to createStore takes 600 ms, did I do something wrong?

### DISCUSSION THREAD

**Discussion by julienrf** [solidjs/solid#1414] [Q&A] [answered]
Hello,
I’ve noticed that when my app is initialized the store creation takes 600 ms:

![Screenshot from 2022-12-10 18-03-35](https://user-images.githubusercontent.com/332812/206866742-e3af5e85-53b0-4a24-881b-4330b1116270.png)

In particular, the call to `hashValue` dominates the execution time.

My call to `createStore` has nothing special, but some properties in my store are large objects. I don’t need _nested_ reactivity for any of those properties. In fact, I only need one level of reactivity. Should I use an object containing signals instead?

**Comment by atk** [comment]
If you want, you can create your own single-layer mutable:
```js
const addMutableProperty(obj, name, initialValue) => {
  const [get, set] = createSignal(initialValue);
  Object.defineProperty(obj, name, { get, set, enumerable: "true" });
}
Object.entries(state).forEach(([key, value]) =>
  addMutableProperty(state, key, value));
```

**Reply by julienrf** [reply]
Thank you for the tip!

What will happen then if I update several fields?

~~~ typescript
state.foo = ...
state.bar = ...
~~~

Is this going to trigger two updates or only one batch update?

**Reply by julienrf** [reply]
Oh, I see, I could use `batch` I guess: https://www.solidjs.com/tutorial/reactivity_batch

**Comment by julienrf** [comment, marked as answer]
I have noticed that if I give the store a “name” the `hashValue` function is not called:

~~~ typescript
createStore(initObject, { name: 'state' });
~~~

I couldn’t find documentation about that. What is the purpose of this `name` option? Is it safe to give it an arbitrary value like this one?

**Reply by atk** [reply]
It's used in the dev tools and to not generate a hash to have a reference for the reactive scope.

**Reply by julienrf** [reply]
Ok, thank you for your answers! Then I think I will go this way because I like the clear separation between getters and setters that stores give you (as opposed to using mutable reactive objects).
